File: chroma-league/src/test/java/com/bonepl/chromaleague/tasks/FetchPlayerNameTaskTest.java
package com.bonepl.chromaleague.tasks;

import com.bonepl.chromaleague.rest.http.LeagueHttpClientMock;
import com.bonepl.chromaleague.state.RunningState;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class FetchPlayerNameTaskTest {

    @AfterEach
    void tearDown() {
        RunningState.setRunningGame(false);
    }

    @Test
    void testPlayerNameFetch() {
        //given
        new LeagueHttpClientMock().mockPlayerNameResponse("json/playername.json");

        //when
        new FetchPlayerNameTask().run();

        //then
        assertEquals("BooonE", RunningState.getGameState().getPlayerName());
    }
}

File: test/tasks/FetchPlayerNameTask.test.ts
import {afterEach, beforeEach, describe, expect, test, vi} from 'vitest';
import {RunningState} from '../../src/state/RunningState.js';

vi.mock('../../src/rest/http/LeagueHttpClients.js', () => ({
    getNonBlockingResponse: vi.fn().mockResolvedValue('BooonE'),
    getRetryingResponse: vi.fn().mockResolvedValue('BooonE'),
    setNonBlockingLeagueHttpClient: vi.fn(),
    setRetryingLeagueHttpClient: vi.fn(),
}));

describe('FetchPlayerNameTask', () => {
    beforeEach(() => {
        RunningState.setRunningGame(false);
        RunningState.setRunningGame(true);
    });

    afterEach(() => {
        RunningState.setRunningGame(false);
    });

    test('testPlayerNameFetch', async () => {
        //given & when
        const { fetchPlayerNameTask } = await import('../../src/tasks/FetchPlayerNameTask.js');
        await fetchPlayerNameTask();

        //then
        expect(RunningState.getGameState().playerRiotId).toBe('BooonE');
    });
});
